pkg/actors/placement/grpc_opts.go

```diff
@@ -45,17 +46,19 @@ func getGrpcOptsGetter(servers []string, clientCert *daprCredentials.CertChain)
 			return cached, nil
 		}
 
-		opts, err := daprCredentials.GetClientOptions(clientCert, security.TLSServerName)
+		var opts []grpc.DialOption
+		placementID, err := spiffeid.FromSegments(sec.ControlPlaneTrustDomain(), "ns", sec.ControlPlaneNamespace(), "dapr-placement")
 		if err != nil {
 			log.Errorf("%s: %v", errEstablishingTLSConn, err)
 			return nil, errEstablishingTLSConn
 		}
 
+		opts = append(opts, sec.GRPCDialOptionMTLS(placementID))
+
 		if diag.DefaultGRPCMonitoring.IsEnabled() {
 			opts = append(
 				opts,
 				grpc.WithUnaryInterceptor(diag.DefaultGRPCMonitoring.UnaryClientInterceptor()),
-				grpc.WithBlock(),
```

Comment:
Why was this removed? This is quite important

Reply:
It is explicitly called out as not recommended in [the docs](https://pkg.go.dev/google.golang.org/grpc#WithBlock), but can add it back if you think we should?

Reply:
It is a fairly significant change actually. We use DialContext to connect to the gRPC server (placement), and the behavior of DialContext [changes](https://pkg.go.dev/google.golang.org/grpc#DialContext) depending on whether WithBlock is set or not:

> In the non-blocking case, the ctx does not act against the connection. It only controls the setup steps.
>
> In the blocking case, ctx can be used to cancel or expire the pending connection. Once this function returns, the cancellation and expiration of ctx will be noop. Users should call ClientConn.Close to terminate all the pending operations after this function returns. 

However, now that I see that... this `grpc.WithBlock` should be added to the options in all cases, and not just if monitoring is enabled (so should be outside of the if block)

Reply:
I added `grpc.WithBlock()` to the actor placement client by default.